### ISSUE TITLE
Compile-time Error: control may reach end of non-void function with Xcode 10.2

### ISSUE THREAD

**Issue by Joyreece** [realm/realm-js#2305]
## TL;DR
Thanks @AidenMontgomery.

1. Open the source file at `${RN_PROJ}/node_modules/realm/src/jsc/jsc_value.hpp`.
2. Find the switch-case segment at [THERE](https://github.com/realm/realm-js/blob/master/src/jsc/jsc_value.hpp#L46).
3. Replace the whole switch-case segment with the following code.

```objective-c
    switch (JSValueGetType(ctx, value)) {
        case kJSTypeNull: return "null";
        case kJSTypeNumber: return "number";
        case kJSTypeObject: return "object";
        case kJSTypeString: return "string";
        case kJSTypeBoolean: return "boolean";
        case kJSTypeUndefined: return "undefined";
        case kJSTypeSymbol: return "symbol";
    }
```

## Goals

When I were compiling the React Native Project with Xcode, the compiler prompts an error.

## Expected Results

No error.

## Actual Results

```
/Users/***/***/***/node_modules/realm/src/jsc/jsc_value.hpp:54:1: error: control may reach end of non-void function [-Werror,-Wreturn-type]
```

## Version of Realm and Tooling

- Realm JS SDK Version: 2.25.0
- React Native: 0.58.6
- Node: 10.15.2
- Client OS & Version: iOS 12.2.0

## Temp Solution

1. Open the source code: `${RN_PROJ}/node_modules/realm/src/jsc/jsc_value.hpp`;
2. Add `return "null";` between line 33 and 34.

I know this is not a good solution.
So I am looking forward to the official solution.

**Comment by HengCC** [comment]
@Joyreece  
similar problem, Today i upgrade system version to 10.14.4 and  xcode version to 10.2,
After that, it will not work properly.This problem has been wasted my half-day time, The error message is as follows:
```
The following build commands failed:
	CompileC /Users/xxx/workspace/yqplus_app/ios/build/Build/Intermediates.noindex/RealmJS.build/Debug-iphonesimulator/RealmJS.build/Objects-normal/x86_64/jsc_value.o jsc/jsc_value.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
```

your solution doesn't work for me.

**Comment by HengCC** [comment]
temp solution :

add ` default: return "null";`  between  line 52 and 53,  

```
template<>
inline const char *jsc::Value::typeof(JSContextRef ctx, const JSValueRef &value) {
    switch (JSValueGetType(ctx, value)) {
        case kJSTypeNull: return "null";
        case kJSTypeNumber: return "number";
        case kJSTypeObject: return "object";
        case kJSTypeString: return "string";
        case kJSTypeBoolean: return "boolean";
        case kJSTypeUndefined: return "undefined";
        default: return "null";
    }
    
}
```

this working for me,  waiting for the official solution

**Comment by AidenMontgomery** [comment]
From what I can tell, and I'm probably wrong, there is a new entry in the JSType Enum **kJSTypeSymbol** which is not being handled in the switch statement.
Therefore the switch is not handling all of the possible values returned by **JSValueGetType**

I don't know what the effect of doing this will be, but I think that the solution is as follows...

```c
inline const char *jsc::Value::typeof(JSContextRef ctx, const JSValueRef &value) {
    switch (JSValueGetType(ctx, value)) {
        case kJSTypeNull: return "null";
        case kJSTypeNumber: return "number";
        case kJSTypeObject: return "object";
        case kJSTypeString: return "string";
        case kJSTypeBoolean: return "boolean";
        case kJSTypeUndefined: return "undefined";
        case kJSTypeSymbol: return "symbol";
    }
}
```

This does mean that when/if a new value is added to the Enum again we will see the same issue, unless there is a default added, I just don't know what that should do.

**Comment by sinhpn92** [comment]
I have same isssue :(

**Comment by ericschaal** [comment]
> From what I can tell, and I'm probably wrong, there is a new entry in the JSType Enum **kJSTypeSymbol** which is not being handled in the switch statement.
> Therefore the switch is not handling all of the possible values returned by **JSValueGetType**
> 
> I don't know what the effect of doing this will be, but I think that the solution is as follows...
> 
> ```c
> inline const char *jsc::Value::typeof(JSContextRef ctx, const JSValueRef &value) {
>     switch (JSValueGetType(ctx, value)) {
>         case kJSTypeNull: return "null";
>         case kJSTypeNumber: return "number";
>         case kJSTypeObject: return "object";
>         case kJSTypeString: return "string";
>         case kJSTypeBoolean: return "boolean";
>         case kJSTypeUndefined: return "undefined";
>         case kJSTypeSymbol: return "symbol";
>     }
> }
> ```
> 
> This does mean that when/if a new value is added to the Enum again we will see the same issue, unless there is a default added, I just don't know what that should do.

makes sense

**Comment by nethergrim** [comment]
Isn't it related to https://github.com/realm/realm-js/issues/2246 ?

**Comment by rajivshah3** [comment]
> From what I can tell, and I'm probably wrong, there is a new entry in the JSType Enum kJSTypeSymbol which is not being handled in the switch statement.

Nope, I think you're correct @AidenMontgomery . This change was introduced in iOS 12.3 (see #2246), which was included in the Xcode update that was released today.

**Comment by StevenMasini** [comment]
> > From what I can tell, and I'm probably wrong, there is a new entry in the JSType Enum kJSTypeSymbol which is not being handled in the switch statement.
> 
> Nope, I think you're correct @AidenMontgomery . This change was introduced in iOS 12.3 (see #2246), which was included in the Xcode update that was released today.

Same conclusion here.

Did someone make a Pull Request for this change ?

**Comment by mohammadalijf** [comment]
@StevenMasini yeah i made the Pull Request #2303 but no luck so far. also if anyone can check if PR is working with Xcode version below 10.2 would be great.

**Comment by StevenMasini** [comment]
@mohammadalijf Ok let me check if that work on XCode 10.1.

**Comment by StevenMasini** [comment]
@mohammadalijf 

I succeed to compile on XCode 10.1 with the fix you made on #2303 

Indeed `kJSTypeSymbol ` isn't defined in JavaScriptCore in the iOS SDK 12.1.

```c
#if defined __IPHONE_12_2 || defined __MAC_10_14_4
     case kJSTypeSymbol: return "symbol";
#endif
```

**Comment by 008v** [comment]
same problem

**Comment by mohammadalijf** [comment]
@StevenMasini thank you steven 👍

**Comment by Bautista-Baiocchi-lora** [comment]
> From what I can tell, and I'm probably wrong, there is a new entry in the JSType Enum **kJSTypeSymbol** which is not being handled in the switch statement.
> Therefore the switch is not handling all of the possible values returned by **JSValueGetType**
> 
> I don't know what the effect of doing this will be, but I think that the solution is as follows...
> 
> ```c
> inline const char *jsc::Value::typeof(JSContextRef ctx, const JSValueRef &value) {
>     switch (JSValueGetType(ctx, value)) {
>         case kJSTypeNull: return "null";
>         case kJSTypeNumber: return "number";
>         case kJSTypeObject: return "object";
>         case kJSTypeString: return "string";
>         case kJSTypeBoolean: return "boolean";
>         case kJSTypeUndefined: return "undefined";
>         case kJSTypeSymbol: return "symbol";
>     }
> }
> ```
> 
> This does mean that when/if a new value is added to the Enum again we will see the same issue, unless there is a default added, I just don't know what that should do.

In which file is this code block? I am trying to implement your fix.

**Comment by sercand** [comment]
Thanks to this issue and [2221](https://github.com/realm/realm-js/issues/2221), [2282](https://github.com/realm/realm-js/issues/2282) issues we switched to SQLite which was a longtime dream of our team. It is clear that realm can't move fast enough to catch the ecosystem.

**Comment by rajivshah3** [comment]
@sercand to be fair, the only way this would have been caught before the Xcode 10.2 public release is if someone noticed this in one of the Xcode 10.2 betas (I did notice this, and probably should have made a PR a long time ago). This was not documented in any SDK or Xcode release notes, at least from what I've seen. It was basically "out of the blue" since it was only added to the JavaScriptCore docs and no other documentation/release notes

**Comment by joshuakelly** [comment]
when can we expect a new release guys?

**Comment by iamawaisakram** [comment]
@Joyreece I do not have realm in my node_modules! Any alternatives?

**Comment by Joyreece** [comment]
> @Joyreece I do not have realm in my node_modules! Any alternatives?

Hi, bro. 
You may have already installed it in the global environment.
Use the `npm -g list | grep realm` to check if it is in the global environment.

If it does, I still recommend that install into the project environment with following commands.
```bash
cd <PROJECT_PATH>
npm install --save realm
```
DO NOT miss the `--save` option~ :)

**Comment by edritech93** [comment]
same issue for me

**Comment by ValeriiKov** [comment]
so, WHEN can we expect a new release guys?

**Comment by rajivshah3** [comment]
@ValeriiKov this was fixed in v2.26.1

@kneth I think this issue can be closed
